[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x10838d9]

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x10838d9]

goroutine 893 [running]:
github.com/projectdiscovery/subfinder/v2/pkg/subscraping/sources/securitytrails.(*Source).Run.func1()
	/home/runner/work/subfinder/subfinder/v2/pkg/subscraping/sources/securitytrails/securitytrails.go:69 +0x499
created by github.com/projectdiscovery/subfinder/v2/pkg/subscraping/sources/securitytrails.(*Source).Run in goroutine 849
	/home/runner/work/subfinder/subfinder/v2/pkg/subscraping/sources/securitytrails/securitytrails.go:42 +0x10f


 Current subfinder version v2.6.4

--- a/v2/pkg/subscraping/sources/securitytrails/securitytrails.go
+++ b/v2/pkg/subscraping/sources/securitytrails/securitytrails.go
@@ -12,6 +12,7 @@ import (
 	jsoniter "github.com/json-iterator/go"
 
 	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+	"github.com/projectdiscovery/utils/ptr"
 )
 
 type response struct {
@@ -66,7 +67,7 @@ func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Se
 				resp, err = session.Get(ctx, fmt.Sprintf("https://api.securitytrails.com/v1/scroll/%s", scrollId), "", headers)
 			}
 
-			if err != nil && resp.StatusCode == 403 {
+			if err != nil && ptr.Safe(resp).StatusCode == 403 {
 				resp, err = session.Get(ctx, fmt.Sprintf("https://api.securitytrails.com/v1/domain/%s/subdomains", domain), "", headers)
 			}
 

